core/win32/injector.c

```diff
@@ -878,10 +878,17 @@ dr_inject_process_attach(process_id_t pid, void **data OUT, char **app_name OUT)
         }
     } while (received_initial_debug_event == false);
 
-    process_handle = OpenProcess(PROCESS_ALL_ACCESS, false, (DWORD)pid);
-    if (process_handle == NULL) {
-        return GetLastError();
-    }
+    info->pi.dwProcessId = dbgevt.dwProcessId;
+
+    DuplicateHandle(GetCurrentProcess(), dbgevt.u.CreateProcessInfo.hProcess,
+                    GetCurrentProcess(), &info->pi.hProcess, 0, FALSE,
+                    DUPLICATE_SAME_ACCESS);
+
+    process_handle = info->pi.hProcess;
+
+    info->pi.hThread = CreateRemoteThread(
+        process_handle, NULL, 0, (LPTHREAD_START_ROUTINE)Sleep, (LPVOID)(SIZE_T)INFINITE,
+        CREATE_SUSPENDED, &info->pi.dwThreadId);
```

Comment:
So this thread is present for the entire execution under DR?  That is not ideal as the app will see it if it iterates its threads (plus it called DLL_THREAD_ATTACH at creation but that's hard to avoid -- I wonder if there's a debug API solution to set the PC of a blocked thread such that we don't need this new thread).  OK to not fix for now but at least put a XXX comment about exiting the thread right after we attach for better transparency.

Reply:
Done. 
Is the phrasing OK?

Reply:
LGTM